.github/workflows/integrationtests.yaml

```diff
@@ -2,6 +2,10 @@ name: Integration Tests
 
 on: [ push, pull_request ]
 
+permissions:
```

Comment:
Just curious to know why these permissions needed ? are they for reading/writing github environment credentials.

nit: Maybe we can adda small comment stating why we need to have these permissions.

Reply:
Thanks, added comment and link